hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/IncrementalInputSplits.java

```diff
@@ -447,14 +448,14 @@ private List<MergeOnReadInputSplit> getInputSplits(
                   .filter(logPath -> !logPath.endsWith(HoodieCDCUtils.CDC_LOGFILE_SUFFIX))
                   .collect(Collectors.toList()));
               String basePath = fileSlice.getBaseFile().map(BaseFile::getPath).orElse(null);
-              String sliceLastInstant = fileSlice.getLatestLogFile().isPresent()
-                      ? fileSlice.getLatestLogFile().get().getDeltaCommitTime()
-                      : fileSlice.getBaseInstantTime();
+              String fileSliceLastestInstant = HoodieTimeline.minInstant(fileSlice.getLatestLogFile().map(HoodieLogFile::getDeltaCommitTime)
+                      .orElse(fileSlice.getBaseInstantTime()), endInstant);
```

Comment:
Caution that the max delta commit time may be less than the base instant time, and it could cause data loss after your change, you need to get the min(max_instant(max_delta_instant, base_instant), endInstant).